packages/server/tests/account.test.ts

```diff
@@ -4,36 +4,99 @@ import AdxApi, {
 } from '@adxp/api'
 import * as util from './_util'
 
+const email = 'alice@test.com'
 const username = 'alice.test'
 const password = 'test123'
 
 describe('account', () => {
+  let serverUrl: string
   let client: AdxServiceClient
   let close: util.CloseFn
 
   beforeAll(async () => {
-    const server = await util.runTestServer()
+    const server = await util.runTestServer({ inviteRequired: true })
     close = server.close
-    client = AdxApi.service(server.url)
+    serverUrl = server.url
+    client = AdxApi.service(serverUrl)
   })
 
   afterAll(async () => {
-    await close()
+    if (close) {
+      await close()
+    }
+  })
+
+  let inviteCode: string
+
+  it('creates an invite code', async () => {
+    const res = await client.todo.adx.createInviteCode(
+      {},
+      { useCount: 1 },
+      {
+        headers: { authorization: util.adminAuth() },
+        encoding: 'application/json',
+      },
+    )
+    inviteCode = res.data.code
+    const [host, code] = inviteCode.split('-')
+    expect(host).toBe(new URL(serverUrl).hostname)
+    expect(code.length).toBe(5)
   })
 
   it('serves the accounts system config', async () => {
     const res = await client.todo.adx.getAccountsConfig({})
+    expect(res.data.inviteCodeRequired).toBe(true)
     expect(res.data.availableUserDomains[0]).toBe('test')
     expect(typeof res.data.inviteCodeRequired).toBe('boolean')
   })
 
+  it('fails on invalid usernames', async () => {
+    try {
+      await client.todo.adx.createAccount(
+        {},
+        {
+          email: 'bad-username@test.com',
+          username: 'did:bad-username.test',
+          password: 'asdf',
+          inviteCode,
+        },
+      )
+      throw new Error('Didnt throw')
+    } catch (e) {
+      console.log(e)
+      expect(
+        e instanceof TodoAdxCreateAccount.InvalidUsernameError,
+      ).toBeTruthy()
+    }
+  })
+
+  it('fails on bad invite code', async () => {
+    try {
+      await client.todo.adx.createAccount(
+        {},
+        {
+          email,
+          username,
+          password,
+          inviteCode: 'fake-invite',
+        },
+      )
+      throw new Error('Didnt throw')
+    } catch (e) {
+      console.log(e)
+      expect(
+        e instanceof TodoAdxCreateAccount.InvalidInviteCodeError,
+      ).toBeTruthy()
+    }
```

Comment:
Just a headsup, I think there's a (arguably) more convenient way to write this—something along the lines of:
```js
await expect(createAccount()).rejects.toThrow(TodoAdxCreateAccount.InvalidInviteCodeError);
```

Reply:
yup good looks 👌 we do that elsewhere in the code but for some reason didn't make it in here. (& you can tell i copy/pasted tests 😛)